### ISSUE TITLE
0.80-dev Android build crashes

### ISSUE THREAD

**Issue by StinkerB06** [nesbox/TIC-80#1106]
Please fix!

**Comment by ddelemeny** [comment]
Confirmed. Last working version seems to be http://dev.nesbox.com/master/2020.05.06

```
05-30 18:29:56.256  2719  2719 F libc    : Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0 in tid 2719 (surfaceflinger), pid 2719 (surfaceflinger)                                                                             
05-30 18:29:56.289  2842  2842 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***                                                                                                                                     
05-30 18:29:56.290  2842  2842 F DEBUG   : LineageOS Version: '16.0-20190908-UNOFFICIAL-a3y17lte'                                                                                                                                              
05-30 18:29:56.290  2842  2842 F DEBUG   : Build fingerprint: 'samsung/a3y17ltexx/a3y17lte:8.0.0/R16NW/A320FXXU4CRL1:user/release-keys'                                                                                                        
05-30 18:29:56.290  2842  2842 F DEBUG   : Revision: '0'                                                                                                                                                                                       
05-30 18:29:56.290  2842  2842 F DEBUG   : ABI: 'arm'                                                                                                                                                                                          
05-30 18:29:56.290  2842  2842 F DEBUG   : pid: 2719, tid: 2719, name: surfaceflinger  >>> /system/bin/surfaceflinger <<<                                                                                                                      
05-30 18:29:56.290  2842  2842 F DEBUG   : signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0                                                                                                                                           
05-30 18:29:56.290  2842  2842 F DEBUG   : Cause: null pointer dereference                                                                                                                                                                     
05-30 18:29:56.290  2842  2842 F DEBUG   :     r0  00000000  r1  00000000  r2  00086b66  r3  40000000                                                                                                                                          
05-30 18:29:56.290  2842  2842 F DEBUG   :     r4  00000000  r5  ffd58298  r6  00000001  r7  ffd582b0                                                                                                                                          
05-30 18:29:56.290  2842  2842 F DEBUG   :     r8  00000001  r9  f4c2d780  r10 00000001  r11 f43d0138                                                                                                                                          
05-30 18:29:56.290  2842  2842 F DEBUG   :     ip  f53aacfc  sp  ffd58260  lr  f53849ed  pc  f5d710f8                                                                                                                                          
05-30 18:29:56.312  2842  2842 F DEBUG   :                                                                                                                                                                                                     
05-30 18:29:56.312  2842  2842 F DEBUG   : backtrace:                                                                                                                                                                                          
05-30 18:29:56.312  2842  2842 F DEBUG   :     #00 pc 0006d0f8  /system/lib/libsurfaceflinger.so (android::RE::impl::RenderEngine::create(int, unsigned int)+208)                                                                              
05-30 18:29:56.312  2842  2842 F DEBUG   :     #01 pc 00071227  /system/lib/libsurfaceflinger.so (android::SurfaceFlinger::init()+606)                                                                                                         
05-30 18:29:56.312  2842  2842 F DEBUG   :     #02 pc 000026d1  /system/bin/surfaceflinger (main+232)                                                                                                                                          
05-30 18:29:56.312  2842  2842 F DEBUG   :     #03 pc 0008c839  /system/lib/libc.so (__libc_init+48)                                                                                                                                           
05-30 18:29:56.312  2842  2842 F DEBUG   :     #04 pc 000025a7  /system/bin/surfaceflinger (_start_main+46)                                                                                                                                    
05-30 18:29:56.312  2842  2842 F DEBUG   :     #05 pc 0001a4ef  /system/bin/linker (__dl__ZNSt3__112__hash_tableINS_17__hash_value_typeIjP6soinfoEENS_22__unordered_map_hasherIjS4_NS_4hashIjEELb1EEENS_21__unordered_map_equalIjS4_NS_8equal_t
oIjEELb1EEENS_9allocatorIS4_EEE14__erase_uniqueIjEEjRKT_+90)                                                                                                                                                                                   
05-30 18:29:56.312  2842  2842 F DEBUG   :     #06 pc 00020c43  <anonymous:ffd38000>                                                                                                                                                           
05-30 18:34:37.349  8071  8071 F libc    : Fatal signal 6 (SIGABRT), code -6 (SI_TKILL) in tid 8071 (SDLActivity), pid 8071 (SDLActivity)                                                                                                      
05-30 18:34:37.439  8090  8090 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***                                                                                                                                     
05-30 18:34:37.439  8090  8090 F DEBUG   : LineageOS Version: '16.0-20190908-UNOFFICIAL-a3y17lte'                                                                                                                                              
05-30 18:34:37.439  8090  8090 F DEBUG   : Build fingerprint: 'samsung/a3y17ltexx/a3y17lte:8.0.0/R16NW/A320FXXU4CRL1:user/release-keys'                                                                                                        
05-30 18:34:37.439  8090  8090 F DEBUG   : Revision: '0'                                                                                                                                                                                       
05-30 18:34:37.439  8090  8090 F DEBUG   : ABI: 'arm'                                                                                                                                                                                          
05-30 18:34:37.439  8090  8090 F DEBUG   : pid: 8071, tid: 8071, name: SDLActivity  >>> com.nesbox.tic <<<                                                                                                                                     
05-30 18:34:37.439  8090  8090 F DEBUG   : signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------                                                                                                                                         
05-30 18:34:37.439  8090  8090 F DEBUG   : Abort message: 'thread.cc:2148] No pending exception expected: java.lang.NoSuchMethodError: no static or non-static method "Lorg/libsdl/app/SDLActivity;.nativePermissionResult(IZ)V"'              
05-30 18:34:37.440  8090  8090 F DEBUG   :     r0  00000000  r1  00001f87  r2  00000006  r3  f511c268                                                                                                                                          
05-30 18:34:37.440  8090  8090 F DEBUG   :     r4  00001f87  r5  00001f87  r6  ff997744  r7  0000010c                                                                                                                                          
05-30 18:34:37.440  8090  8090 F DEBUG   :     r8  00000058  r9  ff9977b1  r10 000009f2  r11 ff9977b0                                                                                                                                          
05-30 18:34:37.440  8090  8090 F DEBUG   :     ip  00000041  sp  ff997730  lr  f50a707d  pc  f509ee0a                                                                                                                                          
05-30 18:34:37.632  8090  8090 F DEBUG   :                                           
05-30 18:34:37.632  8090  8090 F DEBUG   : backtrace:                                                                                                                                                                                          
05-30 18:34:37.632  8090  8090 F DEBUG   :     #00 pc 0001ce0a  /system/lib/libc.so (abort+62)                                                                                                                                                 
05-30 18:34:37.632  8090  8090 F DEBUG   :     #01 pc 00356ebf  /system/lib/libart.so (art::Runtime::Abort(char const*)+798)                                                                                                                   
05-30 18:34:37.632  8090  8090 F DEBUG   :     #02 pc 000071c7  /system/lib/libbase.so (android::base::LogMessage::~LogMessage()+502)                                                                                                          
05-30 18:34:37.632  8090  8090 F DEBUG   :     #03 pc 0037cb83  /system/lib/libart.so (art::Thread::AssertNoPendingException() const+854)                                                                                                      
05-30 18:34:37.632  8090  8090 F DEBUG   :     #04 pc 000dd3b9  /system/lib/libart.so (art::ClassLinker::FindClass(art::Thread*, char const*, art::Handle<art::mirror::ClassLoader>)+28)                                                       
05-30 18:34:37.632  8090  8090 F DEBUG   :     #05 pc 0026700f  /system/lib/libart.so (art::JNI::FindClass(_JNIEnv*, char const*)+1018)                                                                                                        
05-30 18:34:37.632  8090  8090 F DEBUG   :     #06 pc 001251c3  /data/app/com.nesbox.tic-5ZvQEHJKslxUMu2FjfnS5w==/lib/arm/libtic80.so                                                                                                          
05-30 18:34:37.632  8090  8090 F DEBUG   :     #07 pc 00125145  /data/app/com.nesbox.tic-5ZvQEHJKslxUMu2FjfnS5w==/lib/arm/libtic80.so (JNI_OnLoad+96)                                                                                          
05-30 18:34:37.632  8090  8090 F DEBUG   :     #08 pc 00238bf5  /system/lib/libart.so (art::JavaVMExt::LoadNativeLibrary(_JNIEnv*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, _jobject*, std:
:__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>*)+2484)                                                                                                                                                       
05-30 18:34:37.632  8090  8090 F DEBUG   :     #09 pc 00002f73  /system/lib/libopenjdkjvm.so (JVM_NativeLoad+230)                                                                                                                              
05-30 18:34:37.632  8090  8090 F DEBUG   :     #10 pc 001066bb  /system/framework/arm/boot.oat (offset 0x106000) (java.lang.Runtime.nativeLoad [DEDUPED]+130)                                                                                  
05-30 18:34:37.632  8090  8090 F DEBUG   :     #11 pc 00161fdd  /system/framework/arm/boot.oat (offset 0x106000) (java.lang.Runtime.loadLibrary0+148)                                                                                          
05-30 18:34:37.632  8090  8090 F DEBUG   :     #12 pc 001668c7  /system/framework/arm/boot.oat (offset 0x106000) (java.lang.System.loadLibrary+86)                                                                                             
05-30 18:34:37.632  8090  8090 F DEBUG   :     #13 pc 00417d75  /system/lib/libart.so (art_quick_invoke_stub_internal+68)                                                                                                                      
05-30 18:34:37.633  8090  8090 F DEBUG   :     #14 pc 003f1433  /system/lib/libart.so (art_quick_invoke_static_stub+222)                                                                                                                       
05-30 18:34:37.633  8090  8090 F DEBUG   :     #15 pc 000a1043  /system/lib/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+154)                                                       
05-30 18:34:37.633  8090  8090 F DEBUG   :     #16 pc 001e8875  /system/lib/libart.so (art::interpreter::ArtInterpreterToCompiledCodeBridge(art::Thread*, art::ArtMethod*, art::ShadowFrame*, unsigned short, art::JValue*)+232)               
05-30 18:34:37.633  8090  8090 F DEBUG   :     #17 pc 001e3551  /system/lib/libart.so (bool art::interpreter::DoCall<false, false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, art::JValue*)+776
)                                                                                                                                                                                                                                              
05-30 18:34:37.633  8090  8090 F DEBUG   :     #18 pc 003ecf33  /system/lib/libart.so (MterpInvokeStatic+130)                                                                                                                                  
05-30 18:34:37.633  8090  8090 F DEBUG   :     #19 pc 0040ac14  /system/lib/libart.so (ExecuteMterpImpl+14612)
05-30 18:34:37.633  8090  8090 F DEBUG   :     #20 pc 0000df0c  /data/app/com.nesbox.tic-5ZvQEHJKslxUMu2FjfnS5w==/oat/arm/base.vdex (org.libsdl.app.SDL.loadLibrary+212)
05-30 18:34:37.633  8090  8090 F DEBUG   :     #21 pc 001c7ec9  /system/lib/libart.so (_ZN3art11interpreterL7ExecuteEPNS_6ThreadERKNS_20CodeItemDataAccessorERNS_11ShadowFrameENS_6JValueEb.llvm.876446455+352)
05-30 18:34:37.633  8090  8090 F DEBUG   :     #22 pc 001cc797  /system/lib/libart.so (art::interpreter::ArtInterpreterToInterpreterBridge(art::Thread*, art::CodeItemDataAccessor const&, art::ShadowFrame*, art::JValue*)+146)
05-30 18:34:37.633  8090  8090 F DEBUG   :     #23 pc 001e353b  /system/lib/libart.so (bool art::interpreter::DoCall<false, false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, art::JValue*)+754)
05-30 18:34:37.633  8090  8090 F DEBUG   :     #24 pc 003ecf33  /system/lib/libart.so (MterpInvokeStatic+130)
05-30 18:34:37.633  8090  8090 F DEBUG   :     #25 pc 0040ac14  /system/lib/libart.so (ExecuteMterpImpl+14612)
05-30 18:34:37.633  8090  8090 F DEBUG   :     #26 pc 0000aeb8  /data/app/com.nesbox.tic-5ZvQEHJKslxUMu2FjfnS5w==/oat/arm/base.vdex (org.libsdl.app.SDLActivity.loadLibraries+20)
05-30 18:34:37.633  8090  8090 F DEBUG   :     #27 pc 001c7ec9  /system/lib/libart.so (_ZN3art11interpreterL7ExecuteEPNS_6ThreadERKNS_20CodeItemDataAccessorERNS_11ShadowFrameENS_6JValueEb.llvm.876446455+352)
05-30 18:34:37.633  8090  8090 F DEBUG   :     #28 pc 001cc797  /system/lib/libart.so (art::interpreter::ArtInterpreterToInterpreterBridge(art::Thread*, art::CodeItemDataAccessor const&, art::ShadowFrame*, art::JValue*)+146)
05-30 18:34:37.633  8090  8090 F DEBUG   :     #29 pc 001e353b  /system/lib/libart.so (bool art::interpreter::DoCall<false, false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, art::JValue*)+754)
05-30 18:34:37.633  8090  8090 F DEBUG   :     #30 pc 003ee39b  /system/lib/libart.so (MterpInvokeVirtualQuick+422)
05-30 18:34:37.633  8090  8090 F DEBUG   :     #31 pc 0040e814  /system/lib/libart.so (ExecuteMterpImpl+29972)
05-30 18:34:37.633  8090  8090 F DEBUG   :     #32 pc 0000b03c  /data/app/com.nesbox.tic-5ZvQEHJKslxUMu2FjfnS5w==/oat/arm/base.vdex (org.libsdl.app.SDLActivity.onCreate+196)
05-30 18:34:37.633  8090  8090 F DEBUG   :     #33 pc 001c7ec9  /system/lib/libart.so (_ZN3art11interpreterL7ExecuteEPNS_6ThreadERKNS_20CodeItemDataAccessorERNS_11ShadowFrameENS_6JValueEb.llvm.876446455+352)
05-30 18:34:37.633  8090  8090 F DEBUG   :     #34 pc 001cc6e3  /system/lib/libart.so (art::interpreter::EnterInterpreterFromEntryPoint(art::Thread*, art::CodeItemDataAccessor const&, art::ShadowFrame*)+82)
05-30 18:34:37.633  8090  8090 F DEBUG   :     #35 pc 003df79b  /system/lib/libart.so (artQuickToInterpreterBridge+890)
05-30 18:34:37.633  8090  8090 F DEBUG   :     #36 pc 0041c2ff  /system/lib/libart.so (art_quick_to_interpreter_bridge+30)
05-30 18:34:37.633  8090  8090 F DEBUG   :     #37 pc 008cbedd  /system/framework/arm/boot-framework.oat (offset 0x384000) (android.app.Activity.performCreate+124)
05-30 18:34:37.633  8090  8090 F DEBUG   :     #38 pc 00659c87  /system/framework/arm/boot-framework.oat (offset 0x384000) (android.app.Instrumentation.callActivityOnCreate+62)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #39 pc 00745b4f  /system/framework/arm/boot-framework.oat (offset 0x384000) (android.app.ActivityThread.performLaunchActivity+2014)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #40 pc 0074a3e9  /system/framework/arm/boot-framework.oat (offset 0x384000) (android.app.ActivityThread.handleLaunchActivity+360)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #41 pc 008d3653  /system/framework/arm/boot-framework.oat (offset 0x384000) (android.app.servertransaction.LaunchActivityItem.execute+410)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #42 pc 00672b45  /system/framework/arm/boot-framework.oat (offset 0x384000) (android.app.servertransaction.TransactionExecutor.executeCallbacks+516)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #43 pc 006728e9  /system/framework/arm/boot-framework.oat (offset 0x384000) (android.app.servertransaction.TransactionExecutor.execute+216)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #44 pc 0073a3e9  /system/framework/arm/boot-framework.oat (offset 0x384000) (android.app.ActivityThread$H.handleMessage+576)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #45 pc 008fcde9  /system/framework/arm/boot-framework.oat (offset 0x384000) (android.os.Handler.dispatchMessage+136)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #46 pc 008ff55f  /system/framework/arm/boot-framework.oat (offset 0x384000) (android.os.Looper.loop+1294)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #47 pc 00744b91  /system/framework/arm/boot-framework.oat (offset 0x384000) (android.app.ActivityThread.main+664)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #48 pc 00417d75  /system/lib/libart.so (art_quick_invoke_stub_internal+68)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #49 pc 003f1433  /system/lib/libart.so (art_quick_invoke_static_stub+222)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #50 pc 000a1043  /system/lib/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+154)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #51 pc 00350aad  /system/lib/libart.so (art::(anonymous namespace)::InvokeWithArgArray(art::ScopedObjectAccessAlreadyRunnable const&, art::ArtMethod*, art::(anonymous namespace)::ArgArray*, art::JValue*, char const*)+52)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #52 pc 00351ef5  /system/lib/libart.so (art::InvokeMethod(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jobject*, _jobject*, unsigned int)+960)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #53 pc 00302c19  /system/lib/libart.so (art::Method_invoke(_JNIEnv*, _jobject*, _jobject*, _jobjectArray*)+40)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #54 pc 0010c1ff  /system/framework/arm/boot.oat (offset 0x106000) (java.lang.Class.getDeclaredMethodInternal [DEDUPED]+110)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #55 pc 009f8feb  /system/framework/arm/boot-framework.oat (offset 0x384000) (com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run+114)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #56 pc 009fef99  /system/framework/arm/boot-framework.oat (offset 0x384000) (com.android.internal.os.ZygoteInit.main+2944)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #57 pc 00417d75  /system/lib/libart.so (art_quick_invoke_stub_internal+68)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #58 pc 003f1433  /system/lib/libart.so (art_quick_invoke_static_stub+222)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #59 pc 000a1043  /system/lib/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+154)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #60 pc 00350aad  /system/lib/libart.so (art::(anonymous namespace)::InvokeWithArgArray(art::ScopedObjectAccessAlreadyRunnable const&, art::ArtMethod*, art::(anonymous namespace)::ArgArray*, art::JValue*, char const*)+52)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #61 pc 003508c9  /system/lib/libart.so (art::InvokeWithVarArgs(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jmethodID*, std::__va_list)+304)
05-30 18:34:37.634  8090  8090 F DEBUG   :     #62 pc 00294a01  /system/lib/libart.so (art::JNI::CallStaticVoidMethodV(_JNIEnv*, _jclass*, _jmethodID*, std::__va_list)+476)
05-30 18:34:37.635  8090  8090 F DEBUG   :     #63 pc 0006ce03  /system/lib/libandroid_runtime.so (_JNIEnv::CallStaticVoidMethod(_jclass*, _jmethodID*, ...)+38)
05-30 18:34:37.635  8090  8090 F DEBUG   :     #64 pc 0006f02b  /system/lib/libandroid_runtime.so (android::AndroidRuntime::start(char const*, android::Vector<android::String8> const&, bool)+462)
05-30 18:34:37.635  8090  8090 F DEBUG   :     #65 pc 0000198d  /system/bin/app_process32 (main+724)
05-30 18:34:37.635  8090  8090 F DEBUG   :     #66 pc 0008c839  /system/lib/libc.so (__libc_init+48)
05-30 18:34:37.635  8090  8090 F DEBUG   :     #67 pc 00001677  /system/bin/app_process32 (_start_main+46)
05-30 18:34:37.635  8090  8090 F DEBUG   :     #68 pc 00018aff  /system/bin/linker (__dl__ZN6soinfoD1Ev+14)
05-30 18:34:37.635  8090  8090 F DEBUG   :     #69 pc 007feb0c  <anonymous:ff19c000>
```

**Comment by nesbox** [comment]
Started crashing after update to SDL 2.0.12

**Comment by ddelemeny** [comment]
Maybe this needs an update too:
https://github.com/nesbox/TIC-80/tree/master/build/android/app/src/main/java/org/libsdl/app

**Comment by nesbox** [comment]
fixed here 59238c9
thanks to all :)
